include/lsst/ctrl/events/EventSystem.h

```diff
@@ -84,26 +77,23 @@ class EventSystem {
     Event* receiveEvent(const std::string& topicName);
     Event* receiveEvent(const std::string& topicName, const long timeout);
 
-    int64_t createOriginatorId();
-    int extractIPId(int64_t identificationId);
-    int extractProcessId(int64_t identificationId);
-    short extractLocalId(int64_t identificationId);
+    LocationID *createOriginatorId();
 
     StatusEvent* castToStatusEvent(Event* event);
     CommandEvent* castToCommandEvent(Event* event);
 
 private:
+    static EventSystem *defaultEventSystem;
```

Comment:
Maybe it's necessary.  I would just ask that you read http://misko.hevery.com/2008/08/17/singletons-are-pathological-liars/